### PR TITLE
[GH-438] Implement dungeon settlement level up. Fix currency assets

### LINKED ISSUE #438: [FEATURE] Level up Dungeon Settlement
## 🚀 Feature

Depends on #436 

In the Dungeon Settlement scene, a user should be able to:
- See the level of the Dungeon Settlement
- See the cost of leveling up
- Level up the Dungeon Settlement

### PR DESCRIPTION
**Pull request by ncontinanza**
> [!CAUTION]
> To be merged with https://github.com/lambdaclass/mirra_backend/pull/656

Closes #438 

## Motivation

We need to level up the Dungeon Settlement, being able to see the costs in the screen.

## Summary of changes

- Implement a DungeonSettlementManager that will handle most of the gameobjects that display information or perform an action in the Dungeon Settlement scene.
- Implement the functionality of the level up button.
- Update gateway.proto with the necessary changes 

## How has this been tested?

1. Go to the DungeonSettlement scene. 
2. You should see 50 blueprints available to spend in the header, and a cost of 100 gold + 50 blueprints to level up. Your settlement's level should be 1. 
3. Click/tap on the level up button. The level should increment in 1, the blueprints amount in the header should decrease to 0 and the supplies' AFK Rate should not be 0 anymore.
4. Try to level up again. You should get a pop up informing that currency is not enough.

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Review by lotuuu — DISMISSED** [review]

File: client/Assets/Scripts/BackendConnection/SocketConnection.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using System.Linq;
 using System.Threading.Tasks;
 using UnityEngine;
+using UnityEngine.Analytics;
```
Comment (tkz00) [inline client/Assets/Scripts/BackendConnection/SocketConnection.cs:11]:
```suggestion
```

**Review by tkz00 — DISMISSED** [review]
Looks good! Just check the suggestion I left!

**Review by tkz00 — CHANGES_REQUESTED** [review]
Shouldn't the afk rate appear also on the popup? Between the parentheses?

Don't mind the broken assets, I haven't downloaded them.

**Comment by ncontinanza** [comment]
> Shouldn't the afk rate appear also on the popup? Between the parentheses?

Yeah, I'll tackle that pop up when I implement the rewards claim

**Comment by tkz00** [comment]
> > Shouldn't the afk rate appear also on the popup? Between the parentheses?
> 
> Yeah, I'll tackle that pop up when I implement the rewards claim

Great! Then You have my approve!

**Review by tkz00 — DISMISSED** [review]

**Review by lotuuu — CHANGES_REQUESTED** [review]
I modified the seeds to make the las dungeon settlement level have no level_up_cost (something likely to happen when we have real data) and the app crashed:

```
Sequence contains no matching element
UnityEngine.Debug:LogError (object)
SocketConnection:AwaitLevelUpDungeonSettlementResponse (byte[],System.Action`1<User>,System.Action`1<string>) (at Assets/Scripts/BackendConnection/SocketConnection.cs:984)
SocketConnection/<>c__DisplayClass52_0:<LevelUpDungeonSettlement>b__0 (byte[]) (at Assets/Scripts/BackendConnection/SocketConnection.cs:958)
NativeWebSocket.WebSocket:DispatchMessageQueue () (at Assets/WebSocket/WebSocket.cs:640)
SocketConnection:Update () (at Assets/Scripts/BackendConnection/SocketConnection.cs:44)
```
Seeds modification: 
Line 297
```
            level_up_costs:
              case level_number do
                20 ->
                  []

                level_number ->
                  [
                    %{currency_id: gold_currency.id, amount: level_number * 100},
                    %{currency_id: blueprints_currency.id, amount: level_number * 50}
                  ]
              end,
```

**Pull request merged**